### ISSUE TITLE
opaque types with members

### ISSUE THREAD

**Issue by erniecohen** [dafny-lang/dafny#866]
3.0 allows opaque types with members, but trying to use them produces boogie errors.
If the implementation was completed so that these could be used and refined (e.g. with datatypes), I would make heavy use of these immediately.

**Comment by RustanLeino** [comment]
Do you have examples of Boogie errors that were not addressed in PR https://github.com/dafny-lang/dafny/pull/862?

Thanks for your feedback. Updating the refinement features had not been a priority.

**Comment by RustanLeino** [comment]
It would be great to have an example repro. We will revisit this issue after issues https://github.com/dafny-lang/dafny/issues/864 and https://github.com/dafny-lang/dafny/issues/857 have been fixed.

**Comment by erniecohen** [comment]
> Do you have examples of Boogie errors that were not addressed in PR #862?

No, my bad. I had thought you had filed an issue, but looked and didn't find it, so I thought I should file one just in case.
